lib/compilers/dotnet.ts

```diff
@@ -76,16 +64,58 @@ class DotNetCompiler extends BaseCompiler {
         this.corerunPath = path.join(this.clrBuildDir, 'corerun');
         this.crossgen2Path = path.join(this.clrBuildDir, 'crossgen2', 'crossgen2');
         this.ilcPath = path.join(this.clrBuildDir, 'ilc-published', 'ilc');
-        this.ilasmPath = path.join(this.clrBuildDir, 'ilasm');
         this.ildasmPath = path.join(this.clrBuildDir, 'ildasm');
-        this.fscPath = path.join(this.sdkBaseDir, this.sdkVersion, 'FSharp', 'fsc.dll');
-        this.vbcPath = path.join(this.sdkBaseDir, this.sdkVersion, 'Roslyn', 'bincore', 'vbc.dll');
-        this.cscPath = path.join(this.sdkBaseDir, this.sdkVersion, 'Roslyn', 'bincore', 'csc.dll');
 
         this.asm = new DotNetAsmParser();
         this.disassemblyLoaderPath = path.join(this.clrBuildDir, 'DisassemblyLoader', 'DisassemblyLoader.dll');
     }
 
+    getCompilerInfo(lang: LanguageKey): DotNetCompilerInfo {
```

Comment:
Can you make this async and use the non-`Sync` accessor here please (it's more "node"-y to do so)